### DISCUSSION TITLE
using sftp

### DISCUSSION THREAD

**Discussion by pcause** [remotely-save/remotely-save#44] [Ideas]
I'd talked about ssh before.  Windows and Linux both support sftp.  If a user has ssh agent and a config file, all the credentials are safely stored.  Yoou can use sftp to get file listing from the remote server and to copy files or whole directories in either direction.  the shell commands plugin has the code to execute a local command and get the output and you can look at that for code to help.

**Comment by fyears** [comment]
I'd like to add sftp support VERY MUCH, but I am sorry that this is technically impossible to implement this (to make it work cross-platform-ly).

See [this](https://github.com/fyears/remotely-save/blob/master/docs/browser_env_no_nodejs.md) for more details.

**Comment by pcause** [comment]
I wasn't suggesting using a javascript library to do sftp.  execute the sftp native platform app.  as i said you can look at the shell commands extension for how to do that.  you can get output of the sftp app which you can use to dtermine file times and you can also get sftp to transfer files or directories in either direction.

yes, there is an issue for android and perhaps iOS.   It would be great to support on all platforms, but at least you can add desktop support.  on Android I use foldersync pro to do bi0directional sync.  syncthing can also work.  and, yes, that can be used on the desktop, but it is a great convenience to have it in obsidian.
